From 77bf786842c18630fe70dd69228922ade1c6aee3 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@linaro.org>
Date: Wed, 2 Feb 2022 16:06:33 +0100
Subject: [PATCH 0137/1695] drivers: clk: clk-stm32mp15: add
 stm32mp_rcc_clk_to_clock_id()

Adds helper functions stm32mp_rcc_clk_to_clock_id() to get the
clock ID of an RCC clock instance.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 871f8238b6f65fab4e7d023dc3e82d958a891472 drivers: clk: clk-stm32mp15: add stm32mp_rcc_clk_to_clock_id()]
Change-Id: Iecc92bab6d24cbb20371fd1e547927ec943c978b
Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>
---
 core/drivers/clk/clk-stm32mp15.c         | 5 +++++
 core/include/drivers/stm32mp1_rcc_util.h | 1 +
 2 files changed, 6 insertions(+)

diff --git a/core/drivers/clk/clk-stm32mp15.c b/core/drivers/clk/clk-stm32mp15.c
index 9aca8f142..fb298856d 100644
--- a/core/drivers/clk/clk-stm32mp15.c
+++ b/core/drivers/clk/clk-stm32mp15.c
@@ -1392,6 +1392,11 @@ struct clk *stm32mp_rcc_clock_id_to_clk(unsigned long clock_id)
 	return clock_id_to_clk(clock_id);
 }
 
+unsigned int stm32mp_rcc_clk_to_clock_id(struct clk *clk)
+{
+	return clk_to_clock_id(clk);
+}
+
 #if (CFG_TEE_CORE_LOG_LEVEL >= TRACE_DEBUG) && defined(CFG_TEE_CORE_DEBUG)
 struct clk_name {
 	unsigned int clock_id;
diff --git a/core/include/drivers/stm32mp1_rcc_util.h b/core/include/drivers/stm32mp1_rcc_util.h
index 29500c72d..273045647 100644
--- a/core/include/drivers/stm32mp1_rcc_util.h
+++ b/core/include/drivers/stm32mp1_rcc_util.h
@@ -12,6 +12,7 @@ void stm32_reset_system(void);
 
 /* Helper from platform RCC clock driver */
 struct clk *stm32mp_rcc_clock_id_to_clk(unsigned long clock_id);
+unsigned int stm32mp_rcc_clk_to_clock_id(struct clk *clk);
 
 #ifdef CFG_STM32MP15_CLK
 /* Export stm32mp1_clk_ops to make it pager resisdent for STM32MP15 */
-- 
2.47.2

